### ISSUE TITLE
[Regression] space-in-parens ignores comments again

### ISSUE THREAD

**Issue by mzgol** [eslint/eslint#4302]
The following code:
```js
function f(/* config */ config, /* config.proxies */ proxies) {
  return [config, proxies];
}
f();
```
now (in ESLint 1.8.0) fails if `space-in-parens` is enabled with a message: "There should be no spaces inside this paren"; it didn't in 1.7.3.

I'd expect it to not ignore the leading comment and treat it as not having a space directly after a paren (since there's no space there).

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Verified in online demo.

**Comment by alberto** [comment]
It doesn't fail if there is no space after the comment:
```js
function f(/* config */config, /* config.proxies */ proxies) {
  return [config, proxies];
}
f();
```

What should we do then?

**Comment by ilyavolodin** [comment]
Ahh.. you are correct. I didn't notice that. Hmm. In that case I think the rule is behaving correctly. @eslint/eslint-team

**Comment by mzgol** [comment]
The idea here was that there should be no space directly after a paren to which this code adheres. On the other hand, most people prefer to put a space between a comment and code wherever it lies in the file. So IMO either it behaves incorrectly or it needs more options.

**Comment by ilyavolodin** [comment]
@mzgol Was this working in the previous version of ESLint?

**Comment by nzakas** [comment]
We can probably call this a bug. It's a bit of a gray area since now we're (correctly) ignoring comments in this rule, so it looks like the space is right next to the paren when we do the check.

**Comment by mzgol** [comment]
@ilyavolodin Yep, it was working fine in 1.7.3, I wrote that in the description of this issue. It broke the Karma build when ESLint 1.8.0 was released & I had to apply a hotfix: https://github.com/karma-runner/karma/commit/dc1bbab071e70227ffaa6230846d86f93ec4b03c

**Comment by nzakas** [comment]
In 1.7.3, we were incorrectly handling comments, so this isn't really a regression - it's a side effect of actually handling comments correctly in this rule.

**Comment by alberto** [comment]
Right now, this behaviour is common (at least) to the following rules: `array-bracket-spacing`, `block-spacing`, `semi-spacing`, `object-curly-spacing`.

If we are going to change this, I think we should also do the same for the other rules.

**Comment by nzakas** [comment]
Agree. Is this just a change in `isSpaceBetweenTokens`?

**Comment by alberto** [comment]
I don't think so. We can't simply ignore adjacent spaces there, since the rules should also enforce spacing before or after the comment if configured to do so. For example, it is my understanding that the code above should report if the rule was configured with `always`  instead of `never`.

**Comment by nzakas** [comment]
Hmm OK. I'd say let's focus on fixing this one first and see what it takes.

**Comment by alberto** [comment]
Working on this
